tests/test_flags.py

```diff
@@ -0,0 +1,332 @@
+        other = TestFlags(one=False, two=True)
+
+        third = ins
+        ins |= other
+        assert ins is third
+        assert ins.value == 0b011
+        assert other.value == 0b010
+
+        other.four = True
+        ins |= other
+        assert ins.value == 0b111
+
+    def test__xor__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        third = ins ^ other
+        assert third.value == 0b011
+        assert third is not ins
+
+        other.one = True
+        third = ins ^ other
+        assert third.value == 0b010
+
+    def test__ixor__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        third = ins
+        ins ^= other
+        assert ins is third
+        assert ins.value == 0b011
+
+        ins.two = False
+        other.one = True
+        ins ^= other
+        assert ins.value == 0b010
+
+    def test__le__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins <= other
+        other.one = True
+        assert ins <= other
+
+        with pytest.raises(
+            TypeError, match="'<=' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            ins <= 4  # type: ignore  # noqa: B015
+
+        other.value = ins.value
+        assert ins <= other
+
+    def test__ge__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins >= other
+        ins.two = True
+        assert ins >= other
+
+        with pytest.raises(
+            TypeError, match="'>=' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            _ = ins >= 4  # type: ignore
+
+        other.value = ins.value
+        assert ins >= other
+
+    def test__lt__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins < other
+        other.one = True
+        assert ins < other
+
+        with pytest.raises(
+            TypeError, match="'<' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            _ = ins < 4  # type: ignore
+
+        other.value = ins.value
+        assert not ins < other
+
+    def test__gt__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins > other
+        ins.two = True
+        assert ins > other
+
+        with pytest.raises(
+            TypeError, match="'>' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            _ = ins > 4  # type: ignore
+
+        other.value = ins.value
+        assert not ins > other
+
+    def test__invert__(self) -> None:
+        ins = TestFlags(one=True)
+        assert ins.value == 0b0001
+        other = ~ins
+        assert ins.value == 0b0001
```

Comment:
Not sure but this seems out of place?

Reply:
https://github.com/DisnakeDev/disnake/pull/616/commits/312d762e43508ef8dd002bc6cb0c1d79196083aa